### PR TITLE
java metrics on azure container apps

### PR DESCRIPTION
**Pull request by sonwan2020**

### CODE REVIEW

File: articles/container-apps/java-metrics.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  --resource-group <RESOURCE_GROUP> \
+  --enable-java-metrics=false
+```
+
+> [!NOTE]
+> The container app restarts when you update java metrics flag.
+
+## View Java Metrics
+
+<TODO: some sample screenshots for java metrics>
```
Comment (yangtony90) [inline articles/container-apps/java-metrics.md:89]: Let's also mention that all customers can see Java metric names on Azure Monitor, with empty data. But they can only see valid data if they have turned on the feature toggle for "--enable-java-metrics".

File: articles/container-apps/java-metrics.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
```
Comment (yangtony90) [inline articles/container-apps/java-metrics.md:90]: Let's have some examples to break down heap memory into memory pools, e.g., Eden space, Survivor space, etc. And mention that for different JVM implementations, customers may see different memory pool names.
And we should also mention if it is possible to set up Azure Monitor alerts based on such memory pools. We don't need to describe the details, if it is possible for customers to do so, let's just insert a link here to guide them to some existing Azure Monitor doc for more details.

File: articles/container-apps/java-metrics.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+| Java | jvm.buffer.memory.usage | Amount of memory used by buffers, such as direct memory | JvmBufferMemoryUsage | bytes |
+| Java | jvm.buffer.memory.limit | Amount of total memory capacity of buffers | JvmBufferMemoryLimit | bytes |
+| Java | jvm.buffer.count | Number of buffers in the memory pool | JvmBufferCount | n/a |
+| Java | jvm.gc.count | Count of JVM garbage collection actions | JvmGcCount | n/a |
+| Java | jvm.gc.duration | Duration of JVM garbage collection actions | JvmGcDuration | milliseconds |
+| Java | jvm.thread.count | Number of executing platform threads | JvmThreadCount | n/a |
+
+## Configure java metrics
+
+### Portal
```
Comment (yangtony90) [inline articles/container-apps/java-metrics.md:37]: There are existing docs with samples on organizing portal and CLI into tabs. We can leverage that format as well.

File: articles/container-apps/java-metrics.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+ms.author: 
+---
+
+# Java metrics for Java apps in Azure Container Apps
+
+JVM (Java Virtual Machine) metrics are critical for monitoring the health and performance of Java applications.
+They provide insights into various aspects of the JVM, such as memory usage, garbage collection, thread count, and more.
+
+## Azure Container Apps collect list of Java metrics
+
```
Comment (yangtony90) [inline articles/container-apps/java-metrics.md:19]: Please consider if it is helpful to put this link on our doc, for customers to have more information if they need to. https://opentelemetry.io/docs/specs/semconv/runtime/jvm-metrics/ But if it is too complicated to explain how our metrics map to the OTel standard, we can ignore the link.

**Comment by craigshoemaker** [comment]
@sonwan2020 - this PR draft is now available in PR #[273890](https://github.com/MicrosoftDocs/azure-docs-pr/pull/273890).

You can now please close this PR.

**Comment by sonwan2020** [comment]
close this PR, new #https://github.com/MicrosoftDocs/azure-docs-pr/pull/273890.

**Pull request closed without merging**